yt_dlp/extractor/tv5mondeplus.py

```diff
@@ -65,8 +72,20 @@ def _real_extract(self, url):
         formats = []
         for video_file in video_files:
             v_url = video_file.get('url')
+            v_type = video_file.get('type')
             if not v_url:
                 continue
+            if v_type == 'application/deferred':
+                d_param = urllib.parse.quote(v_url)
+                headers = {'Authorization': 'Bearer ' + video_file.get('token')}
+                json = self._download_json(
+                    f'https://api.tv5monde.com/player/asset/{d_param}/resolve?condenseKS=true', v_url,
+                    note='Downloading deferred info', headers=headers)
+                v_url = json[0]['url']
+                video_id = self._search_regex(
+                    r'assets/([\d]{9}_[\da-fA-F]{7})/materials', v_url, 'video id',
+                    default=display_id)
```

Comment:
Is there a reason why we want to extract video id from the deferred video URL instead of from the webpage? Can you explain please

Reply:
the data-guid from the webpage isn't stable for deferred video it seems, so I thought using the asset id from the deferred json.

Reply:
I added a comment about it.

Reply:
Gotcha, but shouldn't we `default=None` here and then `default=display_id` under `if not video_id` later?

Reply:
you're probably right. So remove default=display_id from _search_regex()?